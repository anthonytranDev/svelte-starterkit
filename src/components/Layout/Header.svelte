<script>
  import { Link } from 'svelte-routing'
</script>

<style>
  span {
    font-weight: 700;
  }

  .header {
    padding: 1rem 0.5rem;
  }

  .header-wrapper {
    display: flex;
    justify-content: space-between;
  }

  .header__nav {
    display: inline-flex;
  }

  .header-item {
    padding: 1rem 0.75rem;
  }
</style>

<div class="header" data-testid="Header">
  <div class="header-wrapper">
    <Link to="/">
      <div class="header-item">
        <span>Basic</span> Starterkit
      </div>
    </Link>

    <nav class="header__nav">
      <Link to="/">
        <div class="header-item">Home</div>
      </Link>
      <Link to="/about">
        <div class="header-item">About</div>
      </Link>
      <Link to="/blog">
        <div class="header-item">Blog</div>
      </Link>
    </nav>
  </div>
</div>
