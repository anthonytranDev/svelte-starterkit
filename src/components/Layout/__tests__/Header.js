import '@testing-library/jest-dom/extend-expect'
import { render } from '@testing-library/svelte'
import { Router } from 'svelte-routing'

import Header from '../Header.svelte'

describe('Header', () => {
  it('should render', () => {
    const { getByTestId } = render(
      <Router>
        <Header />
      </Router>
    )

    expect(getByTestId('Header')).toMatchSnapshot()
  })
})
