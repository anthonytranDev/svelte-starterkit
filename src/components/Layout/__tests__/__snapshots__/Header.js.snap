// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render 1`] = `
<div
  class="header svelte-1l7lnf1"
  data-testid="Header"
>
  <div
    class="header-wrapper svelte-1l7lnf1"
  >
    <a
      aria-current="page"
      href="/"
    >
      <div
        class="header-item svelte-1l7lnf1"
      >
        <span
          class="svelte-1l7lnf1"
        >
          Basic
        </span>
         Starterkit
      </div>
    </a>
     
    <nav
      class="header__nav svelte-1l7lnf1"
    >
      <a
        aria-current="page"
        href="/"
      >
        <div
          class="header-item svelte-1l7lnf1"
        >
          Home
        </div>
      </a>
       
      <a
        href="/about"
      >
        <div
          class="header-item svelte-1l7lnf1"
        >
          About
        </div>
      </a>
       
      <a
        href="/blog"
      >
        <div
          class="header-item svelte-1l7lnf1"
        >
          Blog
        </div>
      </a>
    </nav>
  </div>
</div>
`;
